packages/web/docs/src/pages/docs/gateway/other-features/plugins.mdx

```diff
@@ -0,0 +1,576 @@
+- `info`: the GraphQL operation info
+- `fetchFn`: the fetch function that will be used to make the request. By default, it is the one
+  provided in the Gateway configuration if provided, or `@whatwg-node/fetch` if not.
+- `setFetchFn`: replace the `fetch` function that will be used to make the request. It should be
+  compatible with standard `fetch` API.
+- `executionRequest`: present only if the request is an upstream subgraph request. It contains all
+  information about the upstream query, notably the target subgraph name.
+- `requestId`: a uniq id identifying the client request. This is used to correlate downstream and
+  upstream request across services.
+- `logger`: The logger instance for the specific request that includes the details of the request
+  and the response.
+
+### Request Lifecycle
+
+The following diagram shows the lifecycle of request in Hive Gateway.
+
+Each hook is given a single parameter called payload. Please see detail section of each hook to know
+what contains each of this payloads.
+
+```mermaid
+stateDiagram-v2
+  onRequest: <a href="onrequest">onRequest</a>
+  GraphQLRequest: GraphQL Request
+
+  [*] --> onRequest
+
+  state onRequest_if_state <<choice>>
+
+  onRequest --> onRequest_if_state
+
+  onRequest_if_state --> onResponse: Is not a GraphQL Request
+  onRequest_if_state --> GraphQLRequest: Is GraphQL Request
+
+  state GraphQLRequest {
+    onRequestParse: <a href="#onrequestparse">onRequestParse</a>
+    onParams: <a href="#onparams">onParams</a>
+    onEnveloped: <a href="#onenveloped">onEnveloped</a>
+    onParse: <a href="#onparse">onParse</a>
+    onValidate: <a href="#onvalidate">onValidate</a>
+    onContextBuilding: <a href="oncontextbuilding">onContextBuilding</a>
+    onExecute: <a href="#onexecute">onExecute</a>
+    onSubscribe: <a href="#onsubscribe">onSubscribe</a>
+
+    onExecuteDone: <a href="#onexecutedone"> onExecuteDone</a>
+    onSubscribeDone: <a href="#onsubscribedone">onSubscribeDone</a>
+    onExecutionResult: <a href="#onexecutionresult">onExecutionResult</a>
+    onResultProcess: <a href="#onresultprocess">onResultProcess</a>
+
+
+    [*] --> onRequestParse
+    onRequestParse --> onParams
+    onParams --> onEnveloped
+    onEnveloped --> onParse
+    onParse --> onValidate
+    onValidate --> onContextBuilding
+    onContextBuilding --> onExecute
+    onContextBuilding --> onSubscribe
+
+    onExecute --> SubgraphDelegation
+    onSubscribe --> SubgraphDelegation
+
+    SubgraphDelegation: Subgraph Delegation
+    state SubgraphDelegation {
+      onDelegationPlan: <a href="#ondelegationplan">onDelegationPlan</a>
+      onDelegationStageExecute: <a href="#ondelegationstageexecute">onDelegationStageExecute</a>
+      onSubgraphExecute: <a href="#onsubgraphexecute">onSubgraphExecute</a>
+      onSubgraphExecuteDone: <a href="#onsubgraphexecutedone">onSubgraphExecuteDone</a>
+
+      [*] --> onDelegationPlan
+      onDelegationPlan --> onDelegationStageExecute
```

Comment:
Not sure if worth to mention but plan can be memoized so this hook is not always called.

Reply:
Ho, so only the `onDelegationStageExecute` is mandatory ?

If it's cached, `onDelegationPlan` is not called at all ?

Reply:
>Ho, so only the onDelegationStageExecute is mandatory ?

It is always called

>If it's cached, onDelegationPlan is not called at all ?

No it is not if I don't miss anything.